src/debugger.ts

```diff
@@ -150,8 +152,12 @@ export async function getDebugConfigurationFromCache(cache: CMakeCache, target:
 
   if (!debuggerPathOverride) {
     const clang_compiler_regex = /(clang[\+]{0,2})+(?!-cl)/gi;
-    // Look for lldb-mi
-    let clang_debugger_path = compiler_path.replace(clang_compiler_regex, 'lldb-mi');
+    // Look for lldb-mi where the CppTools extension usually installs it
+    // and if not found, try also the compiler path.
+    const cpptoolsExtension = vscode.extensions.getExtension('ms-vscode.cpptools');
+    let clang_debugger_path = cpptoolsExtension ?
```

Comment:
I think the test should be procedural.  Try to see if there is a lldb-mi that matches the compiler.  If not, then try to grab the one from cpptools as a last resort.

Reply:
Ok, makes sense. What's your opinion on the searching for gdb and lldb (not -mi)? Anything you think doesn't make sense now in the search logic and I can remove it now since I'm at it?